ssl/record/rec_layer_s3.c

```diff
@@ -759,24 +759,31 @@ int do_ssl3_write(SSL *s, int type, const unsigned char *buf,
     /* Clear our SSL3_RECORD structures */
     memset(wr, 0, sizeof wr);
     for (j = 0; j < numpipes; j++) {
-        /* write the header */
-        *(outbuf[j]++) = type & 0xff;
-        SSL3_RECORD_set_type(&wr[j], type);
+        if (BIO_get_offload_tx(s->wbio)) {
+            /* IV will be generated by the kernel if offload is used */
+            eivlen = 0;
 
-        *(outbuf[j]++) = (s->version >> 8);
-        /*
-         * Some servers hang if initial client hello is larger than 256 bytes
-         * and record version number > TLS 1.0
-         */
-        if (SSL_get_state(s) == TLS_ST_CW_CLNT_HELLO
-            && !s->renegotiate && TLS1_get_version(s) > TLS1_VERSION)
-            *(outbuf[j]++) = 0x1;
-        else
-            *(outbuf[j]++) = s->version & 0xff;
+            SSL3_RECORD_set_type(&wr[j], type);
+        } else {
+            /* write the header */
+            *(outbuf[j]++) = type & 0xff;
```

Comment:
We seem to do this work twice. Once here if not offloading and the target buffer is outbuf[j]. Then we do it again later if we are offloading but that time the target buffer is the dummy "header". It seems more sensible to just do this once, but change the target buffer as appropriate.